### PR TITLE
Update to Ktor 2.0.0

### PR DESCRIPTION
**Pull request by nhachicha**
This PR adds support for Ktor 2.

This is a breaking change as Ktor 2 only works with the new memory model. So for that reason, we are instead maintaining a separate release branch `release/ktor2-support` with the intent of making this the main branch once the new memory model is announced as stable and the default.

Whenever we make a release from `master`, we should make a similar release from the ktor2 branch.

Changes from `master` should otherwise manually be merged from `release/ktor2-support`.

### CODE REVIEW

**Review by clementetb — APPROVED** [review]
Looking good to me 💎

**Review by edualonso — APPROVED** [review]

File: packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/dynamic/DynamicMutableRealmImpl.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ import io.realm.kotlin.internal.runIfManaged
 import io.realm.kotlin.internal.toRealmObject
 import io.realm.kotlin.query.RealmQuery
 
-internal open class DynamicMutableRealmImpl(
+public open class DynamicMutableRealmImpl(
```
Comment (rorbech) [inline packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/dynamic/DynamicMutableRealmImpl.kt:36]: I always found it nice with a comment that tracked why we exposed those internal classes. Maybe really obvious here, but ...

**Review by rorbech — COMMENTED** [review]
I think there is some communication around this that needs to be a bit more strict, but otherwise LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,8 @@
 
 ### Compatibility
 * This release is compatible with:
+  * The new Kotlin Native memory model only: https://github.com/JetBrains/kotlin/blob/master/kotlin-native/NEW_MM.md
+  * Ktor 2.0.3.
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply CHANGELOG.md:18]: Yes, good idea

File: Jenkinsfile [hunk 3]
```diff
@@ -19,11 +19,16 @@ import groovy.json.JsonOutput
 @Library('realm-ci') _
 
 // Branches from which we release SNAPSHOT's. Only release branches need to run on actual hardware.
-releaseBranches = [ 'master', 'releases', 'next-major' ]
+releaseBranches = [ 'master', 'releases', 'next-major', 'nh/ktor_2.0.0' ]
 // Branches that are "important", so if they do not compile they will generate a Slack notification
 slackNotificationBranches = [ 'master', 'releases', 'next-major' ]
 // Shortcut to current branch name that is being tested
 currentBranch = env.BRANCH_NAME
+if (env.BRANCH_NAME.startsWith('PR-')) {
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply Jenkinsfile:27]: Much better work-around. I agree.

**Pull request merged**